### PR TITLE
filter_add_messages_wrong_behaviour

### PR DESCRIPTION
**Pull request by stefanoj3**
Hello again @harikt 
I was playing with Filter class and noticed a strange behaviour, so I wrote couple of tests to show what are my findings.

Can you confirm that this is a bug? the tests that you see are actually NOT failing
If so I would be happy to provide a fix for it and the tests to cover it (partially here, just need to be cleaned up from comments)

### CODE REVIEW

**Comment by harikt** [comment]
Hi @stefanoj3 ,

You are absolutely correct. This looks like a bug. If you can send a patch I will merge and release it when I get some time.

Thank you.

**Comment by stefanoj3** [comment]
@harikt the PR is updated, also the commit message is now more descriptive
Let me know if I need to adjust something, I tried to follow the standards you follow already in your codebase

Do you plan to tag a 1.2.* version with the fix?

File: tests/Aura/Input/FilterTest.php [hunk 1]
```diff
@@ -140,4 +140,31 @@ function ($value) use ($filter) {
         ];
         $this->assertSame($expect, $actual);
     }
+
```
Comment (harikt) [inline tests/Aura/Input/FilterTest.php:143]: can you please add one more test to add only a string.

**Comment by harikt** [comment]
@stefanoj3 

> Let me know if I need to adjust something, I tried to follow the standards you follow already in your codebase

Thank you very much. I will look more closely in the weekend.

> Do you plan to tag a 1.2.* version with the fix?

Sure I will try to do a release in the weekend.

Thank you for the fix.

**Comment by stefanoj3** [comment]
@harikt PR updated with the additional test you requested 😄

**Comment by harikt** [comment]
Thank you.

**Comment by harikt** [comment]
Released 1.2.1 : https://groups.google.com/d/msg/auraphp/7bZ1Bqd5kbc/gMB8yveGDwAJ

**Pull request merged**